File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.java
// class Solution {
//     public int findMin(int[] nums) {
//         int l = 0;
//         int r = nums.length-1;
//         while(l<r){
//             int m = (l+r)/2;
//             if(nums[m] > nums[l]){
//                 l = m+1;
//             }
//             else{
//                 r = m-1;
//             }
//         }
//         return nums[l];
//     }
// }
class Solution {
    public int findMin(int[] nums) {
        int l = 0;
        int r= nums.length-1;
        while(l<=r){
            if(nums[l]<= nums[r]){
                return nums[l];
            }
            int m = (l+r)/2;
            if(nums[m] >= nums[l]){
                l = m+1;
            }
            else{
                r= m;
            }
        }
        return 0;
    }
}

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l = 0
        r = len(nums)-1
        curr = float("inf")
        while l <=r:
            m = (l+r)//2
            curr = min(curr,nums[m])
            if nums[m] > nums[r]:
                l = m+1
            else:
                r = m-1
        return min(curr,nums[l])
            